core/src/test/java/org/jobrunr/server/tasks/AbstractTaskTest.java

```diff
@@ -60,6 +61,10 @@ protected void setUpBackgroundJobServerConfiguration(BackgroundJobServerConfigur
         // hook to override settings
     }
 
+    protected CarbonAwareJobManager setUpCarbonAwareJobManager() {
+        return null;
+    }
```

Comment:
Should this not always return a mock already? Or even better, a spy() so we can almost end to end test this?

Reply:
I wrote these tests at a time, it was still doing HTTP requests in the background. I'll update and try the spy approach.